### ISSUE TITLE
how to check it 

### ISSUE THREAD

**Issue by danielhop75** [naudio/NAudio#952]
Hi Friends,
I tried to find information if naudio support reading ogg audio files. These files are in fact very popular under linux distro.
Please let me know, put these info to other people if it is possible to read/play ogg files. 
I'd like to know if it is possible in Linux/Android/ and Windows platform. Please put these info into readme of naudio.
Regards,
Daniel

**Comment by markheath** [comment]
Yes, NAudio can read Ogg Vorbis (and Opus), and as of **NAudio 3** there's a proper cross-platform answer. Here's the platform-by-platform picture:

**Windows** — works today. On Windows 10+, `MediaFoundationReader` decodes `.ogg` (Vorbis) directly, since Media Foundation gained Ogg support. The pure-managed **NAudio.Vorbis** package (NVorbis-backed) also works and has no native dependency.

**Linux / macOS** — this is what NAudio 3 adds: the new **NAudio.SoundFile** package, backed by [[libsndfile](https://libsndfile.github.io/libsndfile/)](https://libsndfile.github.io/libsndfile/). `SoundFileReader` is a `WaveStream`/`ISampleProvider` that decodes WAV/AIFF/FLAC/**Ogg-Vorbis**/Opus/MP3, and `SoundFileWriter` can encode them too — the first general-purpose cross-platform decoder/encoder in NAudio (there's no Media Foundation off Windows). It targets `net9.0` with no OS restriction; you just need a system libsndfile installed (`sudo apt install libsndfile1`, `brew install libsndfile`). Codec availability depends on how libsndfile was built — query it with `SoundFileCapabilities.IsFormatSupported(...)`.

```csharp
// reference NAudio.SoundFile, with system libsndfile present
using var reader = new SoundFileReader("audio.ogg");   // WaveStream + ISampleProvider
```

**Android** — this is the gap. `NAudio.SoundFile` P/Invokes a *system* libsndfile, and Android doesn't ship one, so you'd have to bundle a `libsndfile.so` built for each Android ABI and make sure the P/Invoke resolves it. Android isn't an officially tested NAudio target, so no guarantees there. For Android specifically the more realistic option is the **pure-managed NVorbis decoder** (no native dependency), or just using the platform's own `MediaPlayer` for playback.

The format support is documented in the [[NAudio.SoundFile README](https://github.com/naudio/NAudio/blob/main/src/NAudio.SoundFile/README.md)](https://github.com/naudio/NAudio/blob/main/src/NAudio.SoundFile/README.md). Closing this out — the short version is: Ogg works on Windows (Media Foundation / NVorbis) and on Linux/macOS via the new NAudio.SoundFile package; Android needs a bundled native lib or the managed NVorbis route. Happy to help if you hit a specific platform snag.

**Comment by rampaa** [comment]
>On Windows 10+, MediaFoundationReader decodes .ogg (Vorbis) directly, since Media Foundation gained Ogg support.

I think this is not quite right. When I try to use `StreamMediaFoundationReader` to play .ogg (Vorbis) files, I get the following exception: `System.Runtime.InteropServices.COMException: 'The byte stream type of the given URL is unsupported. (0xC00D36C4)'`

Using `VorbisWaveReader` does work as expected. I am on Windows 10 22H2, in case it matters.. If `StreamMediaFoundationReader` just worked and we did not need an additional dependency for .ogg support (namely `NAudio.Vorbis`) that would be nicer.

**Comment by rampaa** [comment]
Note that `Windows Media Player` *is* able to play .ogg (Vorbis) files while `StreamMediaFoundationReader` gives the aforementioned exception. If opening a new issue for this is preferred I can do that as well.

**Comment by mdcdi1315** [comment]
> Note that Windows Media Player is able to play .ogg (Vorbis) files while StreamMediaFoundationReader gives the aforementioned exception. If opening a new issue for this is preferred I can do that as well.

Is `MediaFoundationReader` able read the file or not?
Because if `MediaFoundationReader` is unable to do so, probably there is missing support in Media Foundation, which in such case there is not something that we can do.

**Comment by markheath** [comment]
Yes, `StreamMediaFoundationReader` can only play things that `MediaFoundationReader` can.
Also, `StreamMediaFoundationReader` is at a disadvantage as it doesn't get to see the original filename, or the MIME type (if downloaded from a URL), so it attempts to discover the format. I'm experimenting now with adding OGG support, and also whether a filename hint on `StreamMediaFoundationReader` would help in this situation

**Comment by markheath** [comment]
OK, I have a draft PR to improve this situation. I want to do a bit more testing before I merge it, but hopefully it will help with this situation and be generally more flexible going forwards

**Comment by markheath** [comment]
I've been testing the changes, and I've found that Windows 11 can't play .ogg files via Media Foundation unfortunately (see PR #1392 description for more details on this). However, the change to pass in file name or mime-type hints would be genuinely useful, and adding ogg to the probing will be useful if Windows do ever add support. 

The tests also revealed that its possible we don't need `AudioFileFormatFinder` at all as Windows seems to do some of its own content-probing. But will probably leave it in for now pending further investigation.

**Comment by rampaa** [comment]
> adding ogg to the probing will be useful if Windows do ever add support

Doesn't the fact that `Windows Media Player` is able to play it show that Windows does have the support for it already? Is `Windows Media Player` using a different mechanism for playing ogg files?

Also after trying to migrate to NAudio 3.0.1, I started to face the following issue: https://github.com/naudio/Vorbis/issues/20

So, if I am not missing something obvious, effectively there is no way to play `ogg` files on Windows with NAudio 3.0.0+, apart from perhaps bundling `libsndfile` with our app (which has an LGPL license, which might cause problems depending on the license of our app).

**Comment by mdcdi1315** [comment]
> Doesn't the fact that Windows Media Player is able to play it show that Windows does have the support for it already? Is Windows Media Player using a different mechanism for playing ogg files?

I think that the `Windows Media Player` uses WinRT API's to read files (at least as of Windows 10+), *not* Media Foundation. That's the big difference that does not allow playing OGG Vorbis files in MF. I think that Microsoft provides separate downloadable libraries through their Store that support the OGG codec (among others), but those libraries target the `Windows.Media` WinRT API's *only* (as far as I can recall, @markheath please correct me if I am wrong).

**Comment by markheath** [comment]
I'm looking into releasing a new version of NAudio.Vorbis that's compatible with NAudio 3

But no plans to make any other ogg related changes - it's not a very commonly used format - we just offer what MediaFoundation can play (and libsndfile if you are willing to use it)

**Comment by markheath** [comment]
I've released NAudio.Vorbis 3.0.0-preview.1 which is compatible with NAudio 3: https://www.nuget.org/packages/NAudio.Vorbis/3.0.0-preview.1
Let me know if it works for you, and if I hear no problems after a while, I'll publish a full release
